src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/ExpressionEvaluatorFactory.cs

```diff
@@ -0,0 +1,59 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using DevLab.JmesPath;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class ExpressionEvaluatorFactory : IExpressionEvaluatorFactory
+    {
+        private readonly JmesPath _jmesPath;
+        private readonly ExpressionLanguage _defaultExpressionLanguage;
+
+        public ExpressionEvaluatorFactory()
+            : this(new JmesPath(), ExpressionLanguage.JsonPath)
+        {
+        }
+
+        public ExpressionEvaluatorFactory(ExpressionLanguage defaultLanguage)
+            : this(new JmesPath(), defaultLanguage)
+        {
+        }
+
+        public ExpressionEvaluatorFactory(JmesPath jmesPath, ExpressionLanguage defaultLanguage)
+        {
+            _jmesPath = EnsureArg.IsNotNull(jmesPath, nameof(jmesPath));
+            _defaultExpressionLanguage = defaultLanguage;
+        }
+
+        public IExpressionEvaluator Create(Expression expression)
+        {
+            EnsureArg.IsNotEmptyOrWhiteSpace(expression?.Value, nameof(expression.Value));
+
+            var expressionLanguage = expression.Language ?? _defaultExpressionLanguage;
+
+            switch (expressionLanguage)
```

Comment:
Suggestion: If you do Nate's suggestion you can switch to a switch expression which I feel is more concise and a little easier to read: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression

Reply:
Will do

Reply:
Actually, I had tried using switch expressions before and realized I couldn't. It's a function of C# 8.0+ and this project is set to use `netstandard2.0` and `LanguageVersion=7.3`. So if we want to do this I believe that we'd need to update our target framework, which might also mean updating our build pipeline.

Reply:
Nevermind; looks like you can update the language version independently of the target framework.